consensus/cuckoo/consensus.go

```diff
@@ -693,8 +693,11 @@ func (cuckoo *Cuckoo) Prepare(chain consensus.ChainReader, header *types.Header)
 	}
 	header.Difficulty = cuckoo.CalcDifficulty(chain, header.Time, parent)
 	header.Supply = new(big.Int).Set(parent.Supply)
-	header.Quota = new(big.Int).Add(parent.Quota, new(big.Int).SetUint64(chain.Config().GetBlockQuota(header.Number)))
-	header.QuotaUsed = new(big.Int).Set(parent.QuotaUsed)
+	header.Quota = parent.Quota + chain.Config().GetBlockQuota(header.Number)
+	if header.Quota < parent.Quota {
+		panic("quota reaches the upper limit of uint64")
```

Comment:
Is there any possible to be panic here,  if there is big.Int should be more suitable

Reply:
Due to the consideration of overflow, if the result of addition exceeds the max of uint64, the panic will occur.
If the type is big.Int, it seems to be unlimited.

Reply:
Yes, the big.Int is used for avoid overflow. If it can be overflow , big.Int should be suitable for quota

Reply:
If a block satisfy this condition occurs, all the nodes will crash.

Reply:
we should confirm all the panics in code can't be attacked.

Reply:
In fact, I just test that the addition overflow will panic automatically in Golang. Not like that in C/C++. So we need to do the check before to prevent the panics.